rails-commenting.rb

```diff
@@ -7,24 +7,29 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
+# This class is a sub class which inherits from the superclass ApplicationController. It is a controller which has methods which serve to pass data around and calls on .erb files for display.
 class BlogPostsController < ApplicationController
   def index
     # ---2)
+    # Rails is making every entry in the BlogPost database available to the index route through the instance variable posts.
     @posts = BlogPost.all
   end
 
   # ---3)
+  # The show method is part of the default scaffolding which can be created in a Rails application. In this case the instance variable post is assigned a specific entry in hte BlogPost database found by its id.
   def show
     @post = BlogPost.find(params[:id])
   end
 
   # ---4)
+  # The new method is part of the Rails default scaffolding which generates a new entry in the database. It has available to it an instance variable of a new instance of the BlogPost model.
```

Comment:
An important note to distinguish is the difference between the new method and create.  Create will generate a new instance in the database while new will call on a form.